src/main/java/com/icapps/build/gradle/plugin/utils/ShellHelper.kt

```diff
@@ -13,56 +13,66 @@ object ShellHelper {
 
     private const val PATH_KEY = "PATH"
 
-    fun execWithReader(command: String): BufferedReader {
-        val executable = getExecutable(command)
-        val gitLocation = getCommandLocation(executable)
-        val cleanCommand = command.replaceFirst(executable, gitLocation)
-        return executeWithReader(cleanCommand)
+    fun execWithReader(command: Array<String>): BufferedReader {
+        val location = getCommandLocation(command[0])
```

Comment:
Generally speaking it is not 'nice' to perform this 'destructive' modification of a parameter. Better to pass a more generic container (collection, iterable, ...) and then transform that to the array that is passed to ProcessBuilder